odoo/addons/test_impex/tests/test_load.py

```diff
@@ -1255,3 +1255,167 @@ def test_unique_pair(self):
             m.group(2).split(', '),
             ['Value2', 'Value3']
         )
+
+class test_inherits(ImporterCase):
+    """ The import process should only assign a new xid (derived from the
+    childs') if the child is being created and triggers the creation of the
+    parent
+    """
+    model_name = 'export.inherits.child'
+    def get_xid(self, record):
+        return self.env['ir.model.data'].search([
+            ('model', '=', record._name),
+            ('res_id', '=', record.id),
+        ])
+
+    def test_create_no_parent(self):
+        xids_before = self.env['ir.model.data'].search([])
+        r = self.import_(['id', 'value_parent', 'value'], [
+            ['xxx.child', '0', '1'],
+        ])
+        xids_after = self.env['ir.model.data'].search([])
+        rec = self.env[self.model_name].browse(r['ids'])
+
+        self.assertEqual(rec.value_parent, 0)
+        self.assertEqual(rec.value, 1)
+        self.assertEqual(rec.parent_id.value_parent, 0)
+        self.assertEqual(
+            self.get_xid(rec).mapped('complete_name'),
+            ['xxx.child'],
+        )
+        self.assertEqual(
+            self.get_xid(rec.parent_id).mapped('complete_name'),
+            ['xxx.child_export_inherits_parent'],
+        )
+
+    def test_create_parent_no_xid(self):
+        parent = self.env['export.inherits.parent'].create({'value_parent': 0})
+        r = self.import_(['id', 'parent_id/.id', 'value'], [
+            ['xxx.child', str(parent.id), '1'],
+        ])
+        rec = self.env[self.model_name].browse(r['ids'])
+        self.assertEqual(rec.value_parent, 0)
+        self.assertEqual(rec.parent_id, parent)
+
+        self.assertEqual(
+            self.get_xid(rec).mapped('complete_name'),
+            ['xxx.child'],
+        )
+        self.assertEqual(
+            self.get_xid(rec.parent_id).mapped('complete_name'),
+            [],
+            "no xid should be created for the parent"
+        )
+
+    def test_create_parent_with_xid(self):
+        parent = self.env['export.inherits.parent'].create({'value_parent': 0})
+        pid = self.env['ir.model.data'].create({
+            'model': 'export.inherits.parent',
+            'res_id': parent.id,
+            'module': 'xxx',
+            'name': 'parent',
+        })
```

Comment:
Why not
```suggestion
        parent = self.env['export.inherits.parent']._load_records([{
            'xml_id': 'xxx.parent',
            'values': {'value_parent': 0},
        }])
```

Reply:
To avoid any possibility of funky business, the existence of a parent with a xid is a precondition so I'd rather set it up very specifically.

Reply:
> To avoid any possibility of funky business, the existence of a parent with a xid is a precondition so I'd rather set it up very specifically.

I agree, but that's the purpose of `_load_records`, isn't it?